imagebuildah/executor.go

```diff
@@ -744,18 +744,18 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) (image
 	if dest, err := b.resolveNameToImageRef(b.output); err == nil {
 		switch dest.Transport().Name() {
 		case is.Transport.Name():
-			img, err := is.Transport.GetStoreImage(b.store, dest)
+			_, img, err := util.FindImage(b.store, "", b.systemContext, imageID)
 			if err != nil {
 				return imageID, ref, errors.Wrapf(err, "error locating just-written image %q", transports.ImageName(dest))
 			}
-			if len(b.additionalTags) > 0 {
-				if err = util.AddImageNames(b.store, "", b.systemContext, img, b.additionalTags); err != nil {
-					return imageID, ref, errors.Wrapf(err, "error setting image names to %v", append(img.Names, b.additionalTags...))
-				}
-				logrus.Debugf("assigned names %v to image %q", img.Names, img.ID)
+
+			b.additionalTags = append(b.additionalTags, dest.DockerReference().String())
```

Comment:
That name was assigned to the image when it was committed, wasn't it?

Reply:
@nalind There seems be a race in `c/storage` and layers don't have `name` as the property while they get stored, so in concurrent build somehow the reference for `name` gets altered by other builds using the same `blob`.  

I can drop this particular line if its bothering and add a tight locking while tagging as well. But motivation behind this was just a  additional safety check, this might not be needed. I'll verify and remove this if its not needed.